### PR TITLE
clouddrive2: init at 0.7.21

### PR DESCRIPTION
**Pull request by L-Trump**
Clouddrive2 is a multi-cloud drives management tool supporting mounting cloud drives locally. It supports lots of cloud drives in China.

Homepage: https://www.clouddrive2.com

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/cl/clouddrive2/package.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  os = if stdenv.hostPlatform.isDarwin then "macos" else "linux";
+  arch = if stdenv.hostPlatform.isAarch64 then "aarch64" else "x86_64";
+in
+stdenv.mkDerivation rec {
+  pname = "clouddrive2";
+  version = "0.7.21";
+
+  src = fetchurl {
+    url = "https://github.com/cloud-fs/cloud-fs.github.io/releases/download/v${version}/clouddrive-2-${os}-${arch}-${version}.tgz";
+    sha256 =
```
Comment (SuperSandro2000) [inline pkgs/by-name/cl/clouddrive2/package.nix:17]:
```suggestion
    hash =
```

File: pkgs/by-name/cl/clouddrive2/update.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p bash curl coreutils jq common-updater-scripts
+
+latestTag=$(curl -sSfL https://api.github.com/repos/cloud-fs/cloud-fs.github.io/releases/latest | jq -r ".tag_name")
```
Comment (SuperSandro2000) [inline pkgs/by-name/cl/clouddrive2/update.sh:4]:
```suggestion
latestTag=$(curl -sSfL ${GITHUB_TOKEN:+-u ":$GITHUB_TOKEN"} https://api.github.com/repos/cloud-fs/cloud-fs.github.io/releases/latest | jq -r ".tag_name")
```

**Comment by L-Trump** [comment]
@SuperSandro2000 Done. Thanks for your review!

**Comment by L-Trump** [comment]
@Bot-wxt1221 Done. Thanks for your review!

**Comment by L-Trump** [comment]
Sorry to bother everyone, but is there any progress on this PR?

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349204`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>clouddrive2</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>clouddrive2</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>clouddrive2</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>clouddrive2</li>
  </ul>
</details>

File: pkgs/by-name/cl/clouddrive2/package.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    mkdir -p $out/bin
+    mkdir -p $out/opt/clouddrive2
+    cp -r wwwroot "$out/opt/clouddrive2/wwwroot"
+    cp -r clouddrive "$out/opt/clouddrive2/clouddrive"
+    makeWrapper $out/opt/clouddrive2/clouddrive $out/bin/clouddrive
+
+    runHook postInstall
+  '';
+
+  passthru.updateScript = ./update.sh;
```
Comment (GaetanLepage) [inline pkgs/by-name/cl/clouddrive2/package.nix:41]:
```suggestion
  '';
  
  nativeInstallCheckPhaseInputs = [
    versionCheckHook
  ];
  checkVersionProgramArg = [ "--version" ];
  doInstallCheck = true;

  passthru.updateScript = ./update.sh;
```

Can you check if this works please ?

Comment (L-Trump) [reply pkgs/by-name/cl/clouddrive2/package.nix]: It should seem to be `versionCheckProgramArg` here instead.

**Comment by L-Trump** [comment]
@GaetanLepage Done. Looks working well. Thanks for your review!

**Review by GaetanLepage — APPROVED** [review]
LGTM

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349204`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>clouddrive2</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>clouddrive2</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>clouddrive2</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>clouddrive2</li>
  </ul>
</details>

**Pull request merged**